sympy/simplify/radsimp.py

```diff
@@ -160,6 +160,15 @@ def collect(expr, syms, func=None, evaluate=None, exact=False, distribute_order_
 
     collect_const, collect_sqrt, rcollect
     """
+    _eval_collect = getattr(expr, '_eval_collect', None)
+    if _eval_collect is not None:
+        funcval = func
+        evalval = evaluate
+        exactval = exact
+        distval= distribute_order_term
+        return _eval_collect(syms, func=funcval, evaluate=evalval,
+                             exact=exactval, distribute_order_term=distval)
```

Comment:
This seems quite convoluted compared to:
```
        return _eval_collect(syms, func, evaluate, exact, distribute_order_term)
```

Reply:
I did this because I was getting errors in the passed `**kwargs`. I must have had typos I was not picking up. I have reverted to the simpler form and it works.
commit 6f239980